shuup/core/models/_orders.py

```diff
@@ -1106,5 +1163,47 @@ def get_available_payment_methods(self):
             if m.is_available_for(self)
         ]
 
+    # check whether user can be able to change status of order
+    def validate_order_status_change(self, next_status: OrderStatus):
+        valid_statuses = [OrderStatusRole.NONE, OrderStatusRole.INITIAL, OrderStatusRole.PROCESSING,
+                          OrderStatusRole.COMPLETE, OrderStatusRole.CANCELED]
+
+        if not next_status or next_status not in valid_statuses \
+                or next_status == OrderStatusRole.INITIAL:
+            return False
+
+        curr_role = self.status.role
+        # valid follow of change status:
+        # 1. `INITIAL -> PROCESSING -> COMPLETE`
+        # 2. `INITIAL -> PROCESSING -> CANCELED`
+        # 3. `INITIAL -> CANCELED`
+        if curr_role in [OrderStatusRole.COMPLETE, OrderStatusRole.CANCELED] \
+                or (curr_role == OrderStatusRole.PROCESSING and next_status == OrderStatusRole.INITIAL):
+            return False
+
+        return True
+
+    def change_status(self, next_status: OrderStatus, user: User = None, description: str = None):
+        # validate next_status is valid or not
+
+        if not self.validate_order_status_change(next_status):
+            raise InvalidOrderStatusError("Error! Can not change status to this status")
+
+        try:
+            # update new status of oder
+            self.status = next_status
+
+            # create a new OrderStatusHistory entry
+            order_history = OrderStatusHistory(order=self, previous_order_status=self.status, next_order_status=next_status,
+                                               description=description,
+                                               creator=user
+                                               )
+            order_history.save()
+
+            # end OrderStatusHistory creatation
+            order_status_changed.send(type(self), order=self, old_status=old_status, new_status=self.status)
+        except Exception as changeStatusError:
```

Comment:
you should never do such a try/catch like this, too generic, remove this try/catch block